### ISSUE TITLE
add automatic filter to search results with documentation you have currently open

### ISSUE THREAD

**Issue by tmotyl** [TYPO3-Documentation/t3docs-search-indexer#47]
User is using the search box on certian documentation (e.g. changelog or ext:news).
He is putting keyword in the search box.
he expect to have a facet with current documentation preselected and results shown only from this documentation

**Comment by brotkrueml** [comment]
Hmm, I am not sure, if this is the way it should work: Searching directly on "docs.typo3.org/" results from all manuals are displayed. Using from a search box in "TYPO3 Explained" I only get results from this manual.

At least, the context should be given near the search box (above, below, before, after) so this is visible to the user, like "Searching in TYPO3 Explained", "Searching in News" or "Overall search". And no: a placeholder is not accessible.

My 2 cents ;-)

**Comment by linawolf** [comment]
The idea was to have something like the search on github:

![image](https://github.com/TYPO3-Documentation/t3docs-search-indexer/assets/48202465/cd95e33f-c7a2-4fa5-8c40-2ca81ca4a1ee)

So that uppon searching you can pick the context you would like to search in

**Comment by linawolf** [comment]
Search in this manual all versions,
Search in this category (official docs, or extensions, ...)
Search all over

**Comment by brotkrueml** [comment]
So, some JavaScript thingy with an Ajax backend. Okay, I would love to see a prototype for the frontend beforehand.

**Comment by brotkrueml** [comment]
> Search in this manual all versions, Search in this category (official docs, or extensions, ...) Search all over

Searching in all extensions/versions is mostly not what you want. Just my 2 cents ;-)

But it does not hurt.

**Comment by brotkrueml** [comment]
Yup, but the GUI is an important part. And: no user will remember how the manual they is searching for is written correctly to type it manually. So we need something like a list with checkboxes (at least for the manuals) or a suggest thingy.

**Comment by linawolf** [comment]
true, just saying where the manual is not rerendered it keeps working as-is and you can then put in more facets on the search page itsself

**Comment by brotkrueml** [comment]
I am looking forward to this. And as already said, I postpone the local search for the new rendering until it is decided/implemented that this is the way.

**Comment by soee** [comment]
I have spent some time on this task, mostly thinking about concepts and how they could work, as well as discussing on Slack with members of the documentation team. I proposed to create some draft how the new search could look. Here are the drafts:

![image](https://github.com/TYPO3-Documentation/t3docs-search-indexer/assets/1798417/aec9325a-7f03-43cd-8898-a4e610640401)

This is a very general concept at the moment, and there are many edge cases that need to be discussed. I have not included any filtering by versions here - maybe they should also show up in some form (perhaps a separate filter). Anyway, this is how I see it, but it still needs many improvements. From my perspective, this single task should be discussed in the context of a separate/dedicated budget.

What has to be done here:

- we need a technical draft of how the whole search process should look, what filters we will offer, how they will work, etc.
- extend/modify how and what data we are indexing, as we will need more detailed information to be indexed to support extended suggestions.
- implement on the backend side advanced suggestions handling (endpoints to handle requests from the frontend), where we receive filters from the frontend, parse them, and build an ES query to fetch suggestions, then return them to the frontend.
- build a new search interface on the frontend side as a component (search box) which will process data received from the backend, render forms, handle requests and responses from the backend asynchronously.
